src/windows/models/properties_model.py

```diff
@@ -143,407 +147,277 @@ def update_frame(self, frame_number, reload_model=True):
+                str(value),
+                )
+        # Reduce # of clip properties we are saving (performance boost)
+        c.data = {prop_key: c.data[prop_key]}
+        c.save()
+        # Update the preview
+        get_app().window.refreshFrameSignal.emit()
+        self.parent().clearSelection()
+
+    @staticmethod
+    def set_point(points: list, x_pos: int, value: float) -> bool:
+        """Set a keyframe value in list 'points'"""
+        # Sanity check
+        if len(points) < 1 or "co" not in points[0]:
+            log.warning("Cannot interpret points list!")
+            return False
+        for point in sorted(points, key=lambda p: p["co"]["X"]):
+            log.debug("looping points: co.X = %d", point["co"]["X"])
+            if point["co"]["X"] < x_pos:
+                continue
+            if point["co"]["X"] == x_pos:
+                # Found point, Update value if changed
+                if point["co"]["Y"] == value:
+                    return False
+                log.debug(
+                    "updating point: co.X = %d to value: %.3f",
+                    point["co"]["X"], value)
+                point["co"]["Y"] = value
+                return True
+            if point["co"]["X"] > x_pos:
+                # We passed the point, stop looking
+                break
+        # If we haven't returned, no such point
+        log.debug(
+            "Adding new point at co.X = %d, value %0.3f",
+            x_pos, value)
+        points.append({
+            'co': {'X': x_pos, 'Y': value},
+            'interpolation': 1,
+            })
+        points.sort(key=lambda p: p["co"]["X"])
+        return True
+
+    def update_interpolation(self, item, interpolation=-1, curves=[0, 0, 0, 0]):
+        """Change the interpolation type of the current curve segment"""
+        # Determine what will be changed
+        prop_key, prop_data = self.model.item(item.row(), 0).data()
+        item_id, item_type = item.data()
+        c = PropertiesModel.get_query_object(item_id, item_type)
+        # Sanity checks
+        if not c or prop_key not in c.data:
+            log.error(
+                "Can't find property %s for %s %s",
+                prop_key, item_type, item_id)
+            # Can't find correct values to update
+            return
+        if item_type == "color":
+            # Colors have three keyframe lists
+            points_to_update = [
+                c.data[prop_key][color]["Points"]
+                for color in ["red", "green", "blue"]
+                ]
+        else:
+            points_to_update = [c.data[prop_key]["Points"]]
+        # Will return True if any call produces True, else False
+        updated = any([
+            PropertiesModel.set_interpolation(
+                points,
+                prop_data.get("previous_point_x"),
+                prop_data.get("closest_point_x"),
+                interpolation,
+                curves,
+            ) for points in points_to_update])
```

Comment:
Among the untested and/or unproven assumptions in the new code that _really_ need to be tested before we think of merging this: The assumption here is that the current keyframe curve segment's endpoints are always located at `previous_point_x` and `closest_point_x`.

- [ ] Is that always correct? Is it _ever_ really correct, even?
- [ ] Is it (still?) correct when the playhead is positioned _at_ a keyframe point?
- [ ] Is it (still?) correct when the playhead is positioned **beyond** the last defined keyframe point? (What will `closest_point_x` be in those cases? The same as `self.frame_number`?)
- [ ] Are there ever cases where we **can't** rely on `closest_point_x`, and have to look at `self.frame_number` instead,  when setting interpolation parameters?

Reply:
- [ ] Does `set_interpolation()` ever need to _also_ insert a **value** at `closest_point_x`? (Meaning, does `closest_point_x` ever represent a position that **doesn't** already hold an existing keyframe point, but one that needs to be created? Perhaps when the playhead is positioned out beyond the farthest point in the existing list?)

Reply:
Hmm. Tangentially, I'm now wondering if I should **re**-rename `curves` (which I renamed from the supremely unwieldy `interpolation_details`), and instead call it `handles`?

Reply:
This appears to all be OK, _except_ for a special case when the playhead is positioned at frame 1 of the clip. In those cases, `closest_point_x == previous_point_x == 1`, which confuses the `set_interpolation()` points-loop because looks for `previous_point_x`, finds it and sets `handle_right`, then does a `continue` and starts looking for `closest_point_x` — which it's already passed. So, I just have to special-case the two endpoints being the same. 

(In which case, trying to create a _curve_ at all is nonsensical, so I think `set_interpolation` can just bail immediately.)

Reply:
`prev_x == closest_x` fixed with warning log and early return.

And I did rename `curves` to `handles`, which is more descriptive.

Reply:
> * (What will `closest_point_x` be, in the previous three cases? The same as `self.frame_number`? Which for the last case, would be equal to the clip/transition's end frame.)
> * Are there ever cases where we **can't** rely on `closest_point_x`, and have to look at `self.frame_number` instead,  when setting interpolation parameters?

To answer my own questions, when out past the last keyframe point, `closest_point_x` is always equal to the position of that point, and `previous_point_x` is always the previous point. Which is why you need at least 2 keyframe points before interpolation modes become visible/active in the UI.

#### The good
`closest_point_x` will never point to a non-keyframe point. No worries about that.

#### The ugly-ish
It's possible to choose an interpolation mode with the playhead positioned beyond the last keyframe, and that will (somewhat-surprisingly) change the mode of the _previous_ segment, the one between `previous_point_x` and `closest_point_x`, even though the playhead is not positioned between those two points.

(In any other situation, setting interpolation mode changes the interpolation of the curve segment the playhead is currently positioned in — that is, the interpolation mode between `previous_point_x` < `self.frame_number` <= `closest_point_x`. But after the last keyframe, when `self.frame_number` becomes greater than `closest_point_x`, its value and the value of `previous_point_x` **do not change**, so any keyframe/interpolation operations will still affect the previous keyframe curve segment, _not_ the unbounded one in which the playhead is positioned.)

Reply:
I wonder if we should disable/hide interpolation mode once
`self.frame_number` > ...let's call it `last_point_x`? (Which we can easily check because it's the only condition in which `self.frame_number > closest_point_x`.)

The user can still use `Insert keyframe`, or they can still adjust values to implicitly create a new keyframe, but they can't view or change the interpolation mode of the current curve segment because it doesn't have one.